junit/src/main/java/org/kaazing/k3po/junit/rules/SpecificationStatement.java

```diff
@@ -112,10 +122,11 @@ public void evaluate() throws Throwable {
                 }
             }
 
+            // Stefan - not sure what the comment bellow is about. There is no join() method in the statement or rule class.
```

Comment:
> There is no join() method in the statement or rule. Maybe it is an older comment and the join() method was removed ?

I believe this is true.  But I'm not convinced whether it was correct to remove it or not.  

> I think this is already there, in the assertTrue.  

I'm not following where this is?  What I was trying to state was that the comment seems to indicate that the join is needed for cases where K3PO rule is in a [rule chain](http://www.hascode.com/2012/02/ordering-your-junit-rules-using-a-rulechain/).  Specifically, where rules after it (or maybe before depending on your mental ordering) complete before the K3po rule has completed execution.  If we have tests for this then we can safely remove that comment.  If we don't, perhaps we can add them and then safely remove the comment.